analyzer/src/traversal/expressions.rs

```diff
@@ -693,21 +746,36 @@ fn expr_call_builtin_function(
     typ: GlobalMethod,
     name_span: Span,
     args: &Node<Vec<Node<fe::CallArg>>>,
-) -> Result<ExpressionAttributes, SemanticError> {
+) -> Result<ExpressionAttributes, FatalError> {
     let argument_attributes = expr_call_args(Rc::clone(&scope), context, args)?;
     match typ {
         GlobalMethod::Keccak256 => {
             validate_arg_count(context, typ.into(), name_span, args, 1);
             validate_arg_labels(context, args, &[None]);
 
+            // We only need to return with a FatalError here because the error is already reported
+            // as part of a different check on function arguments.
+            let arg_typ = argument_attributes
+                .first()
+                .map(|attr| &attr.typ)
+                .ok_or(FatalError)?;
+
             if !matches!(
-                argument_attributes.first().map(|attr| &attr.typ),
-                Some(Type::Array(Array {
+                arg_typ,
+                Type::Array(Array {
                     inner: Base::Byte,
                     ..
-                }))
+                })
             ) {
-                return Err(SemanticError::type_error());
+                context.fancy_error(
+                    format!(
+                        "`{}` can not be used as a parameter to `keccak(..)`",
+                        arg_typ
+                    ),
+                    vec![Label::primary(args.span, "wrong type")],
+                    vec!["Note: keccak(..) expects a byte array as parameter".into()],
+                );
+                return Err(FatalError);
```

Comment:
This doesn't need to be fatal, because we know the return type. The labeled span should be the first arg.